src/main/java/Duke.java

```diff
@@ -1,10 +1,167 @@
+                    tasks.markTaskAsDone(taskNo);
+                } else if (Parser.commandEquals("delete", input)) {
+                    int taskNo = Parser.getTaskNo(input);
+                    tasks.deleteTask(taskNo);
+                } else if (Parser.commandEquals("deadline", input)) {
+                    if (Parser.hasNoArgs(input)) throw new EmptyDescriptionException();
+                    if (tasks.isFull()) throw new TooManyTasksException();
+
+                    tasks.addDeadline(Parser.getArgs(input));
+                } else if (Parser.commandEquals("event", input)) {
+                    if (Parser.hasNoArgs(input)) throw new EmptyDescriptionException();
+                    if (tasks.isFull()) throw new TooManyTasksException();
+
+                    tasks.addEvent(Parser.getArgs(input));
+                } else if (Parser.commandEquals("todo", input)) {
+                    if (Parser.hasNoArgs(input)) throw new EmptyDescriptionException();
+                    if (tasks.isFull()) throw new TooManyTasksException();
+
+                    tasks.addTodo(Parser.getArgs(input));
+                } else if (Parser.commandEquals("find", input)) {
+                    if (Parser.hasNoArgs(input)) throw new EmptyDescriptionException();
+
+                    tasks.find(Parser.getArgs(input));
+                } else {
+                    throw new UnknownCommandException();
+                }
+
+                // Rewrites the entire file for every update you make here
+                // Probably O(n^2) time where n is the number of tasks but this is the simplest change we can make
+                storage.writeTasks(tasks);
+            } catch (DukeException e) {
+                Ui.printLine();
+                Ui.indent(e.toString());
+                Ui.printLine();
+            } catch (DateTimeParseException e) {
+                Ui.printLine();
+                Ui.indent("It seems that you have entered a format we don't understand. ");
+                Ui.indent("Please use the YYYY-MM-DD format.");
+                Ui.printLine();
+            }
+        }
+    }
+
+    public static void handleCommand(String input) {
+
+    }
+
+
     public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+        new Duke("data/duke.txt").run();
+    }
+
+    // TODO: Refactor this part to be integrated with CLI UI
+    public String getResponse(String input) {
+        String toReturn;
+        try {
+            if (Parser.commandEquals("bye", input)) {
+                Ui.sayGoodbye();
+                toReturn = "Bye!";
+            } else if (Parser.commandEquals("list", input)) {
+                Ui.displayList(tasks);
```

Comment:
i'd assume that this line is redundant? since you wont be printing to the console but rather printing to the GUi